scripts/globals/job_utils/dancer.lua

```diff
@@ -503,6 +502,14 @@ xi.job_utils.dancer.useWaltzAbility = function(player, target, ability, action)
     amtCured = (target:getStat(xi.mod.VIT) + player:getStat(xi.mod.CHR)) * statMultiplier + waltzInfo[3]
     amtCured = math.floor(amtCured * (1.0 + (math.min(50, player:getMod(xi.mod.WALTZ_POTENCY)) / 100)))
     -- TODO: Account for Waltz Potency Received
+    if player:getStatusEffect(xi.effect.CONTRADANCE) then
+        amtCured = amtCured * 2
+        -- slight delay to allow the effect to apply to all targets of divine waltz
+        -- TODO: Remove this workaround via something in cpp core
+        player:timer(100, function(playerArg)
```

Comment:
https://github.com/LandSandBoat/server/pull/4791#discussion_r1443942144

> @[TeoTwawki](https://github.com/TeoTwawki) TeoTwawki [3 weeks ago](https://github.com/LandSandBoat/server/pull/4791#discussion_r1582399744)
> > Unsure what the hold label was added for
 
> point back at own previous comment. was so nobody merged ahead of conferring on if this was acceptable temporarily, exactly how temporary that would be, and who if anybody would be coming back to do it right. Worth mentioning that temporary measures are technical debt, and past iterations of the project added more technical debt than the GDP of some real world nations.

> @claywar is our resident DNC expert I think. If He's ok with this I'll remove the hold I placed.

Reply:
Updated with new logic and re-tested:
![image](https://github.com/LandSandBoat/server/assets/131182600/2a16e106-a33a-483f-af42-1e46c78a1294)